### ISSUE TITLE
Property or method "toJSON" is not defined

### ISSUE THREAD

**Issue by duziaqin** [vuejs/vue-devtools#333]
**TL;DR**:
when we use `Vue.use` to add a plugin, which render a component instance(with `Vue.extend`), and the component $emit events with vue instance as payload. It cause vue warns.
![image](https://cloud.githubusercontent.com/assets/6905704/26281969/4077d1d4-3e39-11e7-85c5-780fadc3a7b2.png)

**version**: 
vue(>2.0.0) vue-devtool(latest)
**demo**:
 [codepen](https://codepen.io/duziaqin/pen/oWQppm) 
(can't reproduce the bug unless u copy the code, start a demo and open vue-devtool)
**bug**:
vue warns: Property or method "toJSON" is not defined
**facts**:
* JSON.stringify would try to call `toJSON` attr when it meets object. [spec](http://www.ecma-international.org/ecma-262/5.1/#sec-15.12.3)
* vue-devtool use JSON.stringify (or [circular-json-es6](https://github.com/yyx990803/circular-json-es6)) to deep copy(I guess) the $emit payload
* vue use Proxy(_renderProxy) to warn `not defined` attr or method during render in `develop mode`
<img width="607" alt="screen shot 2017-05-21 at 15 31 53" src="https://cloud.githubusercontent.com/assets/6905704/26282041/184c4602-3e3b-11e7-8493-dcfa347fa86e.png">

* vue call the `_init` function (which included _renderProxy) in Vue and Vue.extend(produce _Ctor)

 **reason**
* vue-devtool find out a $emit, try to use JSON.stringify, JSON.stringify meet _Ctor attr first than circular reference
* _Ctor[0] has `_renderProxy`, JSON.stringify try to call `toJSON`, cause getHandler to warn.
* it's a rare bug, because it happens only when `_renderPorxy` is reached before circular reference.

but, it's real confusing...

**Comment by posva** [comment]
Hey, sorry for the delay, I tried your demo on an empty index.html page and no errors appears. I guess it has been fixed recently. If not, add a new repro and I'll take a look 🙂

**Comment by duziaqin** [comment]
please try this demo: [to-json-error](http://monadchen.com/bugs/to-json-error/)

**Comment by posva** [comment]
Maybe you need to update your Vue version. Tell me if you manage to boil it down it in a codepen where I can look at the source code

**Comment by d1manson** [comment]
I'm having this issue too. See component linked to above.

**Comment by Elevista** [comment]
@posva @d1manson 
I have this issue too.
Same dependency with package-lock.json and same dev tools version
but only mac chrome display this error. windows chrome shows nothing

**Comment by KaelWD** [comment]
Same here when I have an array of VueComponents in my data object. 

Chrome 60 on windows 10.

**Comment by ElliotPsyIT** [comment]
Any further insights on this 'toJSON' not defined on the instance warning?  Like @Elevista, I'm not seeing this on Windows Chrome (nor IE11) but I am seeing it on Mac Chrome.  I'm using Vue '2.5.13' and see the error either with or without the Vue dev tools installed.

**Comment by davidmoshal** [comment]
+1 for Mac Chrome

**Comment by intermundos** [comment]
+1 for Mac Chrome

**Comment by intermundos** [comment]
Found out that chrome extension Kuker was the reason for this warning.

**Comment by erwer8363** [comment]
thanks intermundos , it is really because of Kuker, i unstall it , now it is ok...
